test/system/src/main/java/io/pravega/test/system/framework/services/PravegaControllerService.java

```diff
@@ -128,10 +128,10 @@ private App createPravegaControllerApp() {
         parameterList.add(element1);
         app.getContainer().getDocker().setParameters(parameterList);
         //set port
-        app.setPorts(Arrays.asList(CONTROLLER_PORT, REST_PORT));
+        app.setPortDefinitions(Arrays.asList(createPortDefinition(CONTROLLER_PORT), createPortDefinition(REST_PORT)));
         app.setRequirePorts(true);
         List<HealthCheck> healthCheckList = new ArrayList<HealthCheck>();
-        healthCheckList.add(setHealthCheck(900, "TCP", false, 60, 20, 0));
+        healthCheckList.add(setHealthCheck(900, "TCP", false, 60, 20, 0, CONTROLLER_PORT));
```

Comment:
don't  we need `REST-PORT` also here?

Reply:
By default marathon uses `portIndex ` of value `0` . The older code used to use portIndex 0, the current code explicitly ensures this grpc port which is used for all failover tests as part of the health check.